### ISSUE TITLE
User Preferences for xplat template working on every supported platform

### ISSUE THREAD

**Issue by sandreas** [AvaloniaUI/Avalonia#10704]
Currently, I'm trying to develop a small Cross Platform application based on the xplat template. Unfortunately, I cannot find a solution for saving user preferences, that does work for every major platform:

- Desktop (Windows, Linux, macOS) - required
- Android - required
- iOS - required
- WASM - nice to have

I unsuccessfully tried different existing solutions:
- [Xam.Plugins.Settings](http://www.nuget.org/packages/Xam.Plugins.Settings) - did not work for NETCORE projects
- [Xamarin.Essentials](https://docs.microsoft.com/xamarin/essentials/index?WT.mc_id=friends-0000-jamont) - successor of the first one, same error
- [SimpleStorage](https://github.com/zauberzeug/SimpleStorage) - no Provider for Desktop
- `<MauiEssentials/>` - Could not install into the project (`Package 'Microsoft.Maui.Essentials.Runtime.any 7.0.59' has a package type 'DotnetPlatform' that is not supported by project`)

I asked for help via Telegram, but unfortunately nobody suggested alternatives to these.

**Describe the solution you'd like**
I cannot think of any App that does not need to store user preferences (e.g. an API URI), so I would love to see a solution that works similar to [`<MauiEssentials>`](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/storage/preferences?view=net-maui-7.0&tabs=android) for every supported platform like this:

- `Preferences.Set`
- `Preferences.Get`
- `Preferences.Remove`
- `Preferences.Clear`
- `Preferences.ContainsKey`

Best case scenario would be a full integration into the `xplat` template, but I could live with an add-in package providing nuget packages for each platform sharing a common interface.

I also considered encrypted / secret storage for tokens and credentials, but for now I would be happy with a simple unencrypted / insecure storage.

**Describe alternatives you've considered**
See above.

**Additional context**

**Comment by j4587698** [comment]
I think you need this #10628

**Comment by sandreas** [comment]
@j4587698 
> I think you need this https://github.com/AvaloniaUI/Avalonia/issues/10628

Thank you for pointing me to this, but I'm not sure about that would solve my issue. While the `IStorageProvider` API would allow me to store files independently and thus implementing a storage mechanism for values myself, the `Preferences` API seems to be a bit different:

- It can only store values of various simple types (`Int32`, `String`, etc.), no large contents
- It is (or at least should be) available without having a `TopLevel` Control for Service injection (Maybe that also is the case for `IStorageProvider`)
- On `Android` (and maybe `iOS`, some native components seem to come into play to store **Shared** and **Secret** Preferences

I would love to have a discussion about this. The[ `Xamarin.Essentials.Preferences` code is here](https://github.com/xamarin/Essentials/tree/main/Xamarin.Essentials/Preferences), that maybe worth taking a look.

**Comment by j4587698** [comment]
If you got a document file.
You can store in any fotmat, like ini, xml, json, database etc.
so you can store any type and use any encryption method

**Comment by sandreas** [comment]
>  so you can store any type and use any encryption method

Yeah sure, I know that. I might sound a little picky, but there are some native mechanisms (like **Preserve settings on uninstall**), that cannot be used easily with that approach. Sorry for bugging you with that, but I think it might be worth checking out, how others do this (Maui, Xamarin)...

**Comment by sandreas** [comment]
Today I started a project for this: https://github.com/sandreas/Avalonia.Preferences
I need help for Android compatibility and lots of other stuff. Every feedback and help is welcome.

**Comment by rubiomiguel06** [comment]
Still no solution for this? I am in the need of exactly what @sandreas described.

@sandreas I've checked your repo, and works well for me in Windows and iOS, but couldn't make it work in Android (Release)

**Comment by sandreas** [comment]
> @sandreas I've checked your repo, and works well for me in Windows and iOS, but couldn't make it work in Android (Release)

Checkout my new approach: https://github.com/sandreas/Avalonia.SimplePreferences

This SHOULD work on every platform, even WASM, but no guarantees. I did not find a suitable solution with C#, so I changed my project to be a native Android App in Kotlin.

**Comment by rubiomiguel06** [comment]
Now I just ended up using Maui Essentials directly. Have just tested on Android Windows and iOS and seems to be working fine. 
Anyways, this requires [Avalonia.Maui](https://github.com/AvaloniaUI/AvaloniaMauiHybrid)

**Comment by sandreas** [comment]
> Anyways, this requires Avalonia.Maui

I don't like the MAUI stuff... but I'm glad it works for you :-)
